src/websockets.rs

```diff
@@ -15,7 +17,7 @@ pub trait WebsocketMessage: std::fmt::Debug {
     fn text(&self) -> Option<&str>;
 
     /// Returns the text (if there is any) of the error
-    fn error_message(&self) -> Option<&str>;
+    fn error_message(&self) -> Option<Cow<str>>;
```

Comment:
Not a fan of `Cow` unless we really need it.

Reply:
I see you need owned because of the `error.to_string()` so I am guessing `Option<String>` is ok since we do `to_owned` anyway down the line.

Reply:
I am kind of wondering if it'd be worth returning something other than a string from this function.

With async_tungstenite we basically only returned the reason from within a close event from this function, so a `str` maybe made some sense.  Looking into `ws_stream_wasm` though it seems there's a whole host of errors that can happen: https://docs.rs/ws_stream_wasm/latest/ws_stream_wasm/enum.WsErr.html.  Not sure if converting those into a string is the best option...

Guess two other options are: 

1. Another associated type on `WebsocketMessage` that lets each websocket impl provide its own set of errors.
2. Creating a `graphql-ws-client` specific `WebsocketError`/`MessageError`/`WebsocketMessageError` type that encompasses all the possibilities of the various websocket clients and then converting into that.

Not sure which I prefer or if actually a `String` is fine (though it was quite nice to avoid having to allocate in the async-tungstenite code).  What do yous think @johanhelsing & @Sytten?

Reply:
> Guess two other options are:
> 
> 1. Another associated type on `WebsocketMessage` that lets each websocket impl provide its own set of errors.
> 2. Creating a `graphql-ws-client` specific `WebsocketError`/`MessageError`/`WebsocketMessageError` type that encompasses all the possibilities of the various websocket clients and then converting into that.
> 
> Not sure which I prefer or if actually a `String` is fine (though it was quite nice to avoid having to allocate in the async-tungstenite code). What do yous think @johanhelsing & @Sytten?

Good ideas.

I'm making a cross-platform web and native game, so as  user, I'd prefer 2 as it helps me reduce the amount of ifdefs in calling code, though I guess it would probably add a bit of extra maintenance and complexity...

Do you think it should be part of this PR, or should I just revert to `String`s?